### DISCUSSION TITLE
Previewer times out

### DISCUSSION THREAD

**Discussion by gwww** [nvim-telescope/telescope.nvim#1335] [Q&A]
Frequently when I open telescope the file list is empty. After some number of seconds, usually less than 10 (a guess), the list of files will populate. Sometimes I get in the preview window that "Previewer has timed out". I'm on nvim 0.5.1 and latest telescope.

Any thoughts on how to debug? Is there useful info to post here?

**Comment by gwww** [comment]
My telescope config: https://github.com/gwww/dotfiles/blob/main/.config/nvim/lua/plugins/telescope.lua

**Comment by kkharji** [comment]
I have the the same issue

**Comment by Conni2461** [comment]
you should both look at `:help telescope.defaults.preview`. You can change the timeout, etc. It was introduced to make ui not block that much.

**Comment by gwww** [comment]
Good suggestion. The default timeout is 250ms. I see delays of up to 10 seconds before the file list is populated.

**Reply by Conni2461** [reply]
>  I see delays of up to 10 seconds before the file list is populated.

I dont see this, and cant reproduce this. I also need more solid information what you are executing. Which picker `find_files`? How much entries. The 250ms timeout is only about the preview. Counter starts when we start loading the file and we timeout the loading if it takes more than 250ms

**Comment by gwww** [comment]
Thanks!

Yes, using `find_files`. Directory that I invoke it in has about 10 files. The directory is mostly Python code. I am running LSP with `pyright`. My telescope config is pretty plain and is posted in previous reply on this thread.

**Comment by gwww** [comment]
Additional info. I'm running on a MacBook, latest MacOS. When I invoke `find_files` the telescope window pops up and the file list is empty. After 5-10 seconds it will populate. Sometimes it populates right away (as I would expect). Its about 50% or so of the time that it takes 5-10 seconds. It seems to me that it occurs only on the first use of telescope in a nvim session. The second and subsequent uses work as expected.

Still hoping for insight here! Could it be that there is some interaction with pyright/LSP using some resource that blocks telescope? I've only noticed this when editing a Python file. Feels related but have no idea what is going on in the background.

**Comment by felippemr** [comment]
I am having similar issues with a project that has about 300K files.
